LocalizationEditor/Providers/LocalizationProvider.swift

```diff
@@ -28,18 +28,27 @@ final class LocalizationProvider {
      - Parameter key: localization string key
      - Parameter value: new value for the localization string
      */
-    func updateLocalization(localization: Localization, key: String, with value: String) {
-        if let existing = localization.translations.first(where: { $0.key == key }), existing.value == value {
+    func updateLocalization(localization: Localization, key: String, with value: String, message: String?) {
+        if let existing = localization.translations.first(where: { $0.key == key }), existing.value == value, existing.message == message {
             Log.debug?.message("Same value provided for \(existing), not updating")
             return
         }
 
-        Log.debug?.message("Updating \(key) in \(value)")
-
-        localization.update(key: key, value: value)
-
-        let data = localization.translations.map { string in
-            "\"\(string.key)\" = \"\(string.value.replacingOccurrences(of: "\"", with: "\\\""))\";"
+        Log.debug?.message("Updating \(key) in \(value) with Message: \(message ?? "No Message.")")
+
+        localization.update(key: key, value: value, message: message)
+
+        let data = localization.translations.map { string -> String in
+            let stringForMessage: String
+            if let newMessage = message {
```

Comment:
This lookis like a bug, saving all the string with the comment of the string currently being updated.

Probably should be

```suggestion
            if let newMessage = string.message {
```